apps/test/util/testUtils.js

```diff
@@ -462,38 +447,46 @@ export function clearTimeoutsBetweenTests() {
  *   suite.  This should normally be 'false' to avoid unneeded impact on the
  *   runtime of the suite, but setting it 'true' is very useful for isolating
  *   the test that's causing related failures.
+ * @param {function} inScope callback function containing the tests to run
+ *   with the body cleanup check in place.
  */
-export function enforceDocumentBodyCleanup({checkEveryTest = false}) {
+export function enforceDocumentBodyCleanup({checkEveryTest = false}, inScope) {
   let initialInnerHTML;
   const beforeFn = checkEveryTest ? beforeEach : before;
   const afterFn = checkEveryTest ? afterEach : after;
 
-  beforeFn(() => {
-    if (!initialInnerHTML) {
-      initialInnerHTML = document.body.innerHTML;
-    }
-    sinon.spy(document.body, 'addEventListener');
-    sinon.spy(document.body, 'removeEventListener');
-  });
+  // The additional describe calls ensure correct ordering of the
+  // before/after steps relative to other test steps.
+  describe('', () => {
+    beforeFn(() => {
+      if (!initialInnerHTML) {
```

Comment:
this check looks like it will prevent `beforeFn` from being called if we are already inside a different call to `enforceDocumentBodyCleanup`. Could be worth noting in a comment